src/main/java/Wolox/training/controllers/BookController.java

```diff
@@ -19,5 +26,42 @@ public String home() {
         return "home";
     }
 
+    // Create
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Book create(@RequestBody Book book) {
+        return (Book)bookRepository.save(book);
+    }
+
+    // Read
+    @GetMapping
+    public Iterable findAll() {
+        return bookRepository.findAll();
+    }
+
+    @GetMapping("/books/{id}")
+    public Book findById(@PathVariable int id) {
+        Optional<Book> book = bookRepository.findById(id);
+        if (!book.isPresent()) {
```

Comment:
to check if the book is present is great, but you can save yourself the if ( in general we want the fewer ifs we can manage ) 
the better way would be to do 
`return bookRepository.findById(id).orElseThrow(() -> new RuntimeException("The book does not exist")));`